runtime/src/main/java/org/corfudb/runtime/Backup.java

```diff
@@ -322,14 +325,22 @@ private void addToTarFile(File tableFile,TarArchiveOutputStream tarOutput) throw
         });
         return streamIdToTableNameMap;
     }
+
     /**
-     * Cleanup the table backup files under the backupDir directory.
+     * Delete all temp backup directories under the system temp directory.
      */
     private void cleanup() {
-        try {
-            FileUtils.deleteDirectory(new File(backupTempDirPath));
-        } catch (IOException e) {
-            log.error("failed to clean up the temporary backup directory {}", backupTempDirPath);
+        File tmpdir = new File(System.getProperty("java.io.tmpdir"));
+        File[] backupDirs = tmpdir.listFiles(file -> file.getName().contains(BACKUP_TEMP_DIR_PREFIX));
+        if (backupDirs != null) {
+            for (File file : backupDirs) {
+                try {
+                    FileUtils.deleteDirectory(file);
```

Comment:
Before you only had one directory, now it seems you admit more than one. Did not see any test in which you create more than one directory and check the cleanup really removes all.

Reply:
This PR deletes all temp directories because I see a bug where for multiple times the backup failed on halfway and cleanup was not run. This caused accumulation of temp directories. For tests, let me add a test which simulates the situation where there are backup temp dirs before backup.

Reply:
Added cleanupTempDirsBeforeBackupRestoreTest in BackupRestoreIT